llama_index/llm_predictor/base.py

```diff
@@ -250,11 +255,21 @@ def stream(self, prompt: Prompt, **prompt_args: Any) -> Tuple[Generator, str]:
             str: The predicted answer.
 
         """
-        if not isinstance(self._llm, OpenAI):
-            raise ValueError("stream is only supported for OpenAI LLMs")
         formatted_prompt = prompt.format(llm=self._llm, **prompt_args)
-        raw_response_gen = self._llm.stream(formatted_prompt)
-        response_gen = _get_response_gen(raw_response_gen)
+
+        handler = StreamingGeneratorCallbackHandler()
+
+        assert isinstance(self._llm, (BaseLLM, BaseChatModel))
+        self._llm.callbacks = [handler]
+
+        assert isinstance(self._llm, (BaseOpenAI, ChatOpenAI))
```

Comment:
So this means streaming will only work for openai models? In the langchain code, the callback handler should work for any LLM that uses their run manager + on_llm_new_token 🤔

Reply:
yeah agreed here, be explicit that streaming only supported for openai llm's atm

Reply:
revised